cylc/flow/scripts/lint.py

```diff
@@ -673,7 +687,7 @@ def list_wrapper(line: str, check: Callable) -> Optional[Dict[str, str]]:
     },
     'U015': {
         'short': (
-            'Deprecated template variables.'),
+            'Deprecated template variables: {suggest}'),
```

Comment:
Rather than have to rely on variable names matching here (`suggest` in this case but `vars` elsewhere), why not just make the check functions return a list of strings, and add the joined list to the end of the `short` message?

Reply:
Allows us to insert multiple variables 

```
msg = check_meta['short'].format(**check)
```

Reply:
Un drafted - I think that my long term goal is to have a better defined check API so that others can add them, but I think I want this fixed sooner, and will worry about that later.